src/parser.rs

```diff
@@ -14,7 +14,6 @@ pub fn kill_node(node: &NixNode) -> Result<NixNode, String> {
     let b = node.replace_with(a);
     let mut new_root = NixNode::new_root(b);
     loop {
-        dbg!("did one iteration of the inner loop");
```

Comment:
if that stmt isn't there, the whole loop can be compacted:
```rust
    while let Some(parent) = new_root.parent() {
        new_root = parent;
    }
```

Reply:
Changed :+1: